[front] - feat(obs): image generation tracing

JulesBelveze opened the pull request:
## Description
Adds observability tracing for image generation and editing MCP tools using Langfuse. The PR instruments both `generate_image` and `edit_image` tools with traces capturing input parameters, token usage metrics, and cost computation. Error cases are also tracked with appropriate metadata. Additionally includes the `gemini-2.5-flash-image` model in legacy pricing to support cost calculations.

<img width="1539" height="648" alt="Screenshot 2025-12-22 at 15 40 43" src="https://github.com/user-attachments/assets/03ebbf63-8751-47b6-8cbe-960df83eef7e" />

## Risk
Low. The changes are purely additive tracing instrumentation that don't alter the existing image generation/editing logic.

## Tests

Locally

## Deploy Plan

Deploy front

JulesBelveze committed:
[mcp_internal_actions] - feature: implement tracing and cost tracking for image generation/editing

 - Added tracing and cost computation for image generation and image editing using Gemini 2.5 Flash model
 - Introduced random UUIDs for trace identifiers and captured detailed usage and cost metrics associated with image generation and editing operations
 - Stored additional metadata and tags for observability and error tracking during image operations

[assistant] - feature: update token pricing with Gemini 2.5 Flash image model

 - Included Gemini 2.5 Flash image model pricing in the legacy model pricing to support cost computation

[agent_mcp_action_resource] - feature: fetch output items by action IDs

 - Implemented a method to fetch Agent MCP Action output items by their action IDs, improving data retrieval for actions

JulesBelveze committed:
[assistant/token_pricing] - fix: update pricing values for gemini-2.5-flash-image model

 - Doubled the input cost and significantly increased the output cost for the gemini-2.5-flash-image model to reflect updated pricing strategy
 - Updated pricing is likely a response to internal cost changes or market conditions

JulesBelveze committed:
[image_generation] - refactor: streamline token cost calculation for image generation

 - Extract cost calculation logic into a standalone function for improved maintainability
 - Utilize MODEL_PRICING directly to avoid type casting issues with non-standard model IDs
 - Simplify token cost tracking by removing redundant code and implementing structured cost information

flvndvd reviewed (commented):
Let's leverage the existing cost brick 🙏

JulesBelveze committed:
[front/lib/actions/mcp_internal_actions] - refactor: streamline image generation error handling and tracing

- Removed unnecessary creation and use of trace IDs in image generation processes
- Refactored error logging to use normalized error messages for consistent error handling
- Cleaned up metadata and tags associated with image generation and editing observations
- Deleted unused fetchOutputItemsByActionIds function to tidy up the codebase

JulesBelveze committed:
[mcp_internal_actions/servers, assistant] - refactor: streamline image generation cost calculation

- Replaced direct access to MODEL_PRICING with `computeTokensCostForUsageInMicroUsd` for image generation cost details
- Adjusted proportional cost breakdown for the input and output costs in USD

[assistant, types] - feature: add support for image model pricing and type

- Introduced a new constant `IMAGE_MODEL_PRICING` for image generation model pricing
- Exported a combined pricing record including current, legacy, and image models
- Defined new type `ImageModelIdType` for image generation model IDs
- Added image model ID to `IMAGE_MODEL_IDS` array and created corresponding type
